receiver/mysqlreceiver/scraper.go

```diff
@@ -66,7 +63,7 @@ func (m *mySQLScraper) start(_ context.Context, host component.Host) error {
 func (m *mySQLScraper) shutdown(context.Context) error {
 	var err error
 	m.stopOnce.Do(func() {
-		err = m.client.Close()
+		err = m.sqlclient.Close()
 	})
```

Comment:
@djaglowski this seems wrong:
1. It is guarantee that shutdown is called only once (so no need to protect with once).
2. If start fails sqlclient is nil and this will crash here.

Probably fix this in a separate PR.